chromium_src/chrome/browser/ui/views/payments/payment_request_dialog_view.cc

```diff
@@ -0,0 +1,35 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/ui/views/payments/payment_request_dialog_view.h"
+
+#include "brave/browser/brave_rewards/checkout_dialog.h"
+#include "brave/components/brave_rewards/common/constants.h"
+#include "brave/components/payments/content/bat_payment_app_factory.h"
+
+namespace payments {
+
+void PaymentRequestDialogView::CloseDialog() {
+  if (BatPaymentAppFactory::IsBatSupportedMethod(request_)) {
+    return;
+  }
+  PaymentRequestDialogView::CloseDialog_ChromiumImpl();
+}
+
+void PaymentRequestDialogView::ShowDialog() {
+  if (BatPaymentAppFactory::IsBatSupportedMethod(request_)) {
```

Comment:
why is any of this in the DialogView? If you want to show a different Dialog you should be doing it with the `PaymentRequestDelegate`